core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -801,43 +780,63 @@ private[spark] class BlockManager(
     }
 
     var blockWasSuccessfullyStored = false
-    var result: PutResult = null
+    var iteratorFromFailedMemoryStorePut: Option[Iterator[Any]] = None
 
     putBlockInfo.synchronized {
```

Comment:
I think this was a carryover from the very old code where we synchronized on the BlockInfo objects themselves instead of using our new lock interface. I'll remove this to clean things up.